cdap-app-templates/cdap-etl/cdap-etl-proto/src/main/java/io/cdap/cdap/etl/proto/v2/validation/StageValidationRequest.java

```diff
@@ -43,6 +54,10 @@ public ETLStage getStage() {
     return inputSchemas == null ? Collections.emptyList() : inputSchemas;
   }
 
+  public boolean getResolveMacrosFromPreferences() {
+    return resolveMacrosFromPreferences;
```

Comment:
since this class is deserialized from user input, this can actually be null. Should check for null and return false if so.

Reply:
Fixed.